src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2151,6 +2151,13 @@ void ProcessInteropMethod(MethodDefinition method)
 		{
 			TypeDefinition returnTypeDefinition = method.ReturnType.Resolve ();
 
+			if (!string.IsNullOrEmpty(_context.PInvokesListFile) && method.IsPInvokeImpl) {
+				_context.PInvokes.Add (new PInvokeInfo {
```

Comment:
This should keep only unique pairs

Reply:
The output must include multiple identical native functions if those functions are from different managed assemblies.